oauth.go

```diff
@@ -0,0 +1,154 @@
+type OAuthScopeType string
+
+// List of possible values for OAuth scopes.
+const (
+	OAuthScopeTypeReadOnly  OAuthScopeType = "read_only"
+	OAuthScopeTypeReadWrite OAuthScopeType = "read_write"
+)
+
+// OAuthTokenType is the type of token. This will always be "bearer."
+type OAuthTokenType string
+
+// List of possible OAuthTokenType values.
+const (
+	OAuthTokenTypeBearer OAuthTokenType = "bearer"
+)
+
+// OAuthStripeUserBusinessType is the business type for the Stripe oauth user.
+type OAuthStripeUserBusinessType string
+
+// List of supported values for business type.
+const (
+	OAuthStripeUserBusinessTypeCorporation OAuthStripeUserBusinessType = "corporation"
+	OAuthStripeUserBusinessTypeLLC         OAuthStripeUserBusinessType = "llc"
+	OAuthStripeUserBusinessTypeNonProfit   OAuthStripeUserBusinessType = "non_profit"
+	OAuthStripeUserBusinessTypePartnership OAuthStripeUserBusinessType = "partnership"
+	OAuthStripeUserBusinessTypeSoleProp    OAuthStripeUserBusinessType = "sole_prop"
+)
+
+// OAuthStripeUserGender of the person who will be filling out a Stripe
+// application. (International regulations require either male or female.)
+type OAuthStripeUserGender string
+
+// The gender of the person who  will be filling out a Stripe application.
+// (International regulations require either male or female.)
+const (
+	OAuthStripeUserGenderFemale OAuthStripeUserGender = "female"
+	OAuthStripeUserGenderMale   OAuthStripeUserGender = "male"
+)
+
+// OAuthError is the type of errors raised when failing authorization.
+type OAuthError string
+
+// List of supported OAuthError values.
+const (
+	OAuthErrorInvalidGrant            OAuthError = "invalid_grant"
+	OAuthErrorInvalidRequest          OAuthError = "invalid_request"
+	OAuthErrorInvalidScope            OAuthError = "invalid_scope"
+	OAuthErrorUnsupportedGrantType    OAuthError = "unsupported_grant_type"
+	OAuthErrorUnsupportedResponseType OAuthError = "unsupported_response_type"
+)
+
+// DeauthorizeError the type of errors raised when failing authorization.
+type DeauthorizeError string
+
+// List of supported DeauthorizeError values.
+const (
+	DeauthorizeErrorInvalidClient  DeauthorizeError = "invalid_client"
+	DeauthorizeErrorInvalidRequest DeauthorizeError = "invalid_request"
+)
+
+// OAuthStripeUserParams for the stripe_user OAuth Authorize params.
+type OAuthStripeUserParams struct {
+	BlockKana          *string `form:"block_kana"`
+	BlockKanji         *string `form:"block_kanji"`
+	BuildingKana       *string `form:"building_kana"`
+	BuildingKanji      *string `form:"building_kanji"`
+	BusinessName       *string `form:"business_name"`
+	BusinessType       *string `form:"business_type"`
+	City               *string `form:"city"`
+	Country            *string `form:"country"`
+	Currency           *string `form:"currency"`
+	DOBDay             *uint64 `form:"dob_day"`
+	DOBMonth           *uint64 `form:"dob_month"`
+	DOBYear            *uint64 `form:"dob_year"`
```

Comment:
And oops, actually if you wouldn't mind, these should be `*int64` actually. They can't be negative so technically `uint64` would be a better fit, but at one point we just decided to just go with `int64` everywhere by convention — in practice we'll never get to numbers high enough that we need the extra range `uint64` gets us, and using `int64` gives us some forward compatibility in case a field that currently doesn't allow negative numbers allows them in the future.